### PR TITLE
feat: Portal UI

### LINKED ISSUE #398: Update Portal UI for Hacker, Mentor, Volunteer
UI needs to be updated according to [Figma](https://www.figma.com/design/CyxJ0OSjgNPwb0eeTgq84s/IrvineHacks-2025?node-id=1-2&node-type=frame&t=kQSodRUO6VFsNpBg-0)

### PR DESCRIPTION
**Pull request by jotalis**
Summary
- Updated the UI for the portal page according to the Figma

Test Plan
- Go to ApplicantPortal.tsx and comment out lines 24-28
- Add "let status = identity.status; status = PortalStatus.pending;" 
- Test each of the possible PortalStatus states by changing what status is set to; e.g. status = PortalStatus.waived;...
- Ensure that the UI looks right

Closes #398

### CODE REVIEW

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/RSVP.tsx
Comment (waalbert) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/RSVP.tsx:1]: Suggestion [repeating]: rename this file to `RSVPComponent` to follow the consistency of `TimelineComponent`

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/Submission.tsx
Comment (waalbert) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/Submission.tsx:1]: Suggestion [repeating]: rename this file to `SubmissionComponent` to follow the consistency of `TimelineComponent`

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/Verdict.tsx
Comment (waalbert) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/Verdict.tsx:1]: Suggestion [repeating]: rename this file to `VerdictComponent` to follow the consistency of `TimelineComponent`

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/Waiver.tsx
Comment (waalbert) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/Waiver.tsx:1]: Suggestion [repeating]: rename this file to `WaiverComponent` to follow the consistency of `TimelineComponent`

**Review by waalbert — CHANGES_REQUESTED** [review]
Awesome work Jay! Here are a few suggestions:

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/TimelineComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		? "text-[var(--color-white)] bg-[var(--color-black)]"
+		: "text-[var(--color-black)] bg-[var(--color-white)]";
+
+	return (
+		<div className={`${baseStyles} ${colorStyles}`}>
+			<h2 className="text-xs sm:text-base md:text-4xl mx-4 sm:mx-8 md:mx-16 my-5 md:my-[70px]">
+				{text}
+			</h2>
+			<StatusImage statusIcon={statusIcon} finished={finished} />
+		</div>
```
Comment (IanWearsHat) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/TimelineComponent.tsx:26]: Issue: There should be some particle assets to the sides of the Timeline.

image

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/TimelineComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	finished: boolean;
+	statusIcon: "Accepted" | "Rejected" | "Pending";
+}
+
+export const TimelineComponent: React.FC<TimelineComponentProps> = ({
+	text,
+	finished,
+	statusIcon,
+}) => {
+	const baseStyles = `flex items-center justify-between mb-4 md:mb-8 border-[2px] md:border-[5px] border-[var(--color-white)]`;
```
Comment (IanWearsHat) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/TimelineComponent.tsx:15]: Nitpick: string should be in double quotes if there is nothing templated

File: apps/site/src/app/(main)/portal/@applicant/components/timeline/VerticalTimeline.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { WaiverComponent } from "./Waiver";
+import { RSVPComponent } from "./RSVP";
+
+interface VerticalTimelineProps {
+	status: PortalStatus;
+}
+
+function VerticalTimeline({ status }: VerticalTimelineProps) {
+	return (
+		<div className="">
```
Comment (IanWearsHat) [inline apps/site/src/app/(main)/portal/@applicant/components/timeline/VerticalTimeline.tsx:14]: Suggestion: empty `className` prop could be removed

**Review by IanWearsHat — CHANGES_REQUESTED** [review]
Great job working on this complex task especially with all of its different statuses. I left a few comments below.

**Review by IanWearsHat — APPROVED** [review]
Looks good to me! Thanks again Jay for working on this complicated task. Squash and merge when ready.

File: apps/site/src/app/(main)/portal/@applicant/components/ReturnHome.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import Button from "@/lib/components/Button/Button";
+
+export default function SignWaiver() {
```
Comment (waalbert) [inline apps/site/src/app/(main)/portal/@applicant/components/ReturnHome.tsx:3]: Issue: `SignWaiver` is already a component defined elsewhere, so naming this `ReturnHome` will reduce confusion

**Review by waalbert — CHANGES_REQUESTED** [review]
One more thing

**Review by waalbert — APPROVED** [review]

**Pull request merged**